docs/ref/models/constraints.txt

```diff
@@ -95,3 +100,8 @@ ensures that each user only has one draft.
 
 These conditions have the same database restrictions as
 :attr:`Index.condition`.
+
+.. note::
+
+    The ``ValidationError`` is raised during model validation when the
+    ``UniqueConstraint`` is violated.
```

Comment:
I think we need to say a little more than just adding these `note` items. 

In general constraints are **not** checked during `full_clean()`, and so do not raise `ValidationErrors`. Rather you'll get a database error on `save()`. `UniqueConstraint`s are different in this regard, in that they leverage the existing `validate_unique()` logic.